ergo/distributions/point_density.py

```diff
@@ -0,0 +1,260 @@
+        else:
+            raise Exception("Point Density does not accept custom xs specifications")
+
+        return super(PointDensity, cls).from_conditions(
+            *args, fixed_params=fixed_params, scale=scale, **kwargs
+        )
+
+    @classmethod
+    def from_params(cls, fixed_params, opt_params, scale=Scale(0, 1), traceable=True):
+        xs = fixed_params["xs"]
+        ps = nn.softmax(opt_params) * opt_params.size
+        densities = ps
+        return cls(
+            xs=xs, densities=densities, scale=scale, normalized=True, traceable=True
+        )
+
+    # Optimize Create Helpers
+
+    @staticmethod
+    def initialize_optimizable_params(fixed_params):
+        num_points = fixed_params["xs"].size
+        return onp.full(num_points, 1.0)
+
+    @classmethod
+    def normalize_fixed_params(self, fixed_params, scale):
+        return {"xs": scale.normalize_points(fixed_params["xs"])}
+
+    # Export
+
+    def to_arrays(self, metaculus_denorm=False):
+        xs = self.scale.denormalize_points(self.normed_xs)
+
+        if metaculus_denorm:
+            # Make sure points cover whole scale, return normed densities
+            densities = self.normed_densities
+            if xs[0] != self.scale.low:
+                density = (densities[0] - densities[1]) / 2 + densities[0]
+                clamped_density = onp.maximum(density, 0)
+
+                xs = onp.append(onp.array([self.scale.low]), xs)
+                densities = onp.append(np.array([clamped_density]), densities)
+            if xs[-1] != self.scale.high:
+                density = (densities[-1] - densities[-2]) / 2 + densities[-1]
+                clamped_density = onp.maximum(density, 0)
+
+                xs = onp.append(xs, onp.array([self.scale.high]))
+                densities = onp.append(densities, np.array([clamped_density]))
+        else:
+            densities = self.scale.denormalize_densities(xs, self.normed_densities)
+
+        return xs, densities
+
+    def to_pairs(self, metaculus_denorm=False):
+        xs, densities = self.to_arrays(metaculus_denorm)
+        pairs = [
+            {"x": float(x), "density": float(density)}
+            for x, density in zip(xs, densities)
+        ]
+        return pairs
+
+    # Condition Methods
+
+    def entropy(self):
+        return -np.dot(self.bin_probs, np.log(self.bin_probs))
+
+    def cross_entropy(self, q_dist):
+        # We assume that the distributions are on the same scale!
+        return -np.dot(self.bin_probs, np.log(q_dist.bin_probs))
+
+    def mean(self):
+        normed_mean = np.dot(self.normed_xs, self.bin_probs)
+        return self.scale.denormalize_point(normed_mean)
+
+    def variance(self):
+        normed_mean = np.dot(self.normed_xs, self.bin_probs)
+        normed_variance = np.dot(
+            self.bin_probs, np.square(self.normed_xs - normed_mean)
+        )
+        return self.scale.denormalize_variance(normed_variance)
```

Comment:
TODO: Revisit this once Dan says it's fixed.

Reply:
Ready for review